GoCD job will fails when there have no pull request in github

The Go-Agent log as follow:

```
2015-12-04 21:19:08,527 [loopThread] ERROR thoughtworks.go.work.DefaultGoPublisher:137 - Interaction with plugin with id 'github.pr' implementing 'scm' extension failed while requesting for 'checkout'. Reason: [Unsuccessful response from plugin. Plugin returned with code '500' and the following response: '"Exception Occurred: [git, clone, --branch\u003dmaster, xxxx.git, /var/lib/go-agent/pipelines/x] - null"']xxx
java.lang.RuntimeException: Interaction with plugin with id 'github.pr' implementing 'scm' extension failed while requesting for 'checkout'. Reason: [Unsuccessful response from plugin. Plugin returned with code '500' and the following response: '"Exception Occurred: [git, clone, --branch\u003dmaster, xxxx.git, /var/lib/go-agent/pipelines/xxxx] - null"']
        at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:38)
        at com.thoughtworks.go.plugin.access.scm.SCMExtension.checkout(SCMExtension.java:175)
        at com.thoughtworks.go.domain.materials.scm.PluggableSCMMaterialAgent.prepare(PluggableSCMMaterialAgent.java:51)
        at com.thoughtworks.go.remote.work.BuildWork.prepareJob(BuildWork.java:168)
        at com.thoughtworks.go.remote.work.BuildWork.build(BuildWork.java:136)
        at com.thoughtworks.go.remote.work.BuildWork.doWork(BuildWork.java:91)
        at com.thoughtworks.go.agent.JobRunner.run(JobRunner.java:53)
        at com.thoughtworks.go.agent.AgentController.retrieveWork(AgentController.java:186)
        at com.thoughtworks.go.agent.AgentController.loop(AgentController.java:132)
        at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.springframework.util.MethodInvoker.invoke(MethodInvoker.java:273)
        at org.springframework.scheduling.support.MethodInvokingRunnable.run(MethodInvokingRunnable.java:65)
        at org.springframework.scheduling.timer.DelegatingTimerTask.run(DelegatingTimerTask.java:70)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
Caused by: java.lang.RuntimeException: Unsuccessful response from plugin. Plugin returned with code '500' and the following response: '"Exception Occurred: [git, clone, --branch\u003dmaster, xxxx.git, /var/lib/go-agent/pipelines/xxxx] - null"'
        at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:36)
        ... 16 more
```

--- a/src/main/java/in/ashwanthkumar/gocd/github/GitHubPRBuildPlugin.java
+++ b/src/main/java/in/ashwanthkumar/gocd/github/GitHubPRBuildPlugin.java
@@ -20,6 +20,7 @@ import in.ashwanthkumar.gocd.github.util.GitFactory;
 import in.ashwanthkumar.gocd.github.util.GitFolderFactory;
 import in.ashwanthkumar.gocd.github.util.JSONUtils;
 import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
 
 import java.io.IOException;
 import java.lang.reflect.Constructor;
@@ -343,7 +344,11 @@ public class GitHubPRBuildPlugin implements GoPlugin {
     }
 
     GitConfig getGitConfig(Map<String, String> configuration) {
-        GitConfig gitConfig = new GitConfig(configuration.get("url"), configuration.get("username"), configuration.get("password"), null);
+        GitConfig gitConfig = new GitConfig(
+                configuration.get("url"),
+                configuration.get("username"),
+                configuration.get("password"),
+                StringUtils.trimToNull(configuration.get("defaultBranch")));
         provider.addConfigData(gitConfig);
         return gitConfig;
     }

--- a/src/main/java/in/ashwanthkumar/gocd/github/settings/scm/DefaultScmPluginConfigurationView.java
+++ b/src/main/java/in/ashwanthkumar/gocd/github/settings/scm/DefaultScmPluginConfigurationView.java
@@ -20,6 +20,7 @@ public class DefaultScmPluginConfigurationView implements ScmPluginConfiguration
         response.put("url", FieldFactory.createForScm("URL", null, true, true, false, "0"));
         response.put("username", FieldFactory.createForScm("Username", null, false, false, false, "1"));
         response.put("password", FieldFactory.createForScm("Password", null, false, false, true, "2"));
+        response.put("defaultBranch", FieldFactory.createForScm("Default Branch", "master", false, false, false, "3"));
         return response;
     }
 

--- a/src/main/resources/views/scm.template.branch.filter.html
+++ b/src/main/resources/views/scm.template.branch.filter.html
@@ -14,6 +14,11 @@
     <span class="form_error" ng-show="GOINPUTNAME[password].$error.server">{{ GOINPUTNAME[password].$error.server }}</span>
 </div>
 <div class="form_item_block">
+    <label>Default Branch:</label>
+    <input type="text" ng-model="defaultBranch" ng-required="false"/>
+    <span class="form_error" ng-show="GOINPUTNAME[defaultBranch].$error.server">{{ GOINPUTNAME[defaultBranch].$error.server }}</span>
+</div>
+<div class="form_item_block">
     <label>Whitelisted brances:</label>
     <input type="text" ng-model="branchwhitelist" ng-required="false"/>
     <span class="form_error" ng-show="GOINPUTNAME[branchwhitelist].$error.server">{{ GOINPUTNAME[branchwhitelist].$error.server }}</span>

--- a/src/main/resources/views/scm.template.html
+++ b/src/main/resources/views/scm.template.html
@@ -13,3 +13,8 @@
     <input type="password" ng-model="password" ng-required="false"/>
     <span class="form_error" ng-show="GOINPUTNAME[password].$error.server">{{ GOINPUTNAME[password].$error.server }}</span>
 </div>
+<div class="form_item_block">
+    <label>Default Branch:</label>
+    <input type="text" ng-model="defaultBranch" ng-required="false"/>
+    <span class="form_error" ng-show="GOINPUTNAME[defaultBranch].$error.server">{{ GOINPUTNAME[defaultBranch].$error.server }}</span>
+</div>

--- a/src/test/java/in/ashwanthkumar/gocd/github/provider/gerrit/GerritProviderTest.java
+++ b/src/test/java/in/ashwanthkumar/gocd/github/provider/gerrit/GerritProviderTest.java
@@ -23,9 +23,9 @@ public class GerritProviderTest extends in.ashwanthkumar.gocd.github.provider.Ab
         PluginConfigurationView scmConfigurationView = getScmView();
 
         assertThat(scmConfigurationView.fields().keySet(),
-                   hasItems("url", "username", "password")
+                   hasItems("url", "username", "password", "defaultBranch")
         );
-        assertThat(scmConfigurationView.fields().size(), is(3));
+        assertThat(scmConfigurationView.fields().size(), is(4));
     }
 
     @Test

--- a/src/test/java/in/ashwanthkumar/gocd/github/provider/git/GitProviderTest.java
+++ b/src/test/java/in/ashwanthkumar/gocd/github/provider/git/GitProviderTest.java
@@ -23,9 +23,9 @@ public class GitProviderTest extends AbstractProviderTest {
         PluginConfigurationView scmConfigurationView = getScmView();
 
         assertThat(scmConfigurationView.fields().keySet(),
-                   hasItems("url", "username", "password", "branchwhitelist", "branchblacklist")
+                   hasItems("url", "username", "password", "branchwhitelist", "branchblacklist", "defaultBranch")
         );
-        assertThat(scmConfigurationView.fields().size(), is(5));
+        assertThat(scmConfigurationView.fields().size(), is(6));
     }
 
     @Test

--- a/src/test/java/in/ashwanthkumar/gocd/github/provider/github/GitHubProviderTest.java
+++ b/src/test/java/in/ashwanthkumar/gocd/github/provider/github/GitHubProviderTest.java
@@ -23,9 +23,9 @@ public class GitHubProviderTest extends AbstractProviderTest {
         PluginConfigurationView scmConfigurationView = getScmView();
 
         assertThat(scmConfigurationView.fields().keySet(),
-                   hasItems("url", "username", "password")
+                   hasItems("url", "username", "password", "defaultBranch")
         );
-        assertThat(scmConfigurationView.fields().size(), is(3));
+        assertThat(scmConfigurationView.fields().size(), is(4));
     }
 
     @Test

--- a/src/test/java/in/ashwanthkumar/gocd/github/provider/stash/StashProviderTest.java
+++ b/src/test/java/in/ashwanthkumar/gocd/github/provider/stash/StashProviderTest.java
@@ -23,9 +23,9 @@ public class StashProviderTest extends AbstractProviderTest {
         PluginConfigurationView scmConfigurationView = getScmView();
 
         assertThat(scmConfigurationView.fields().keySet(),
-                   hasItems("url", "username", "password")
+                   hasItems("url", "username", "password", "defaultBranch")
         );
-        assertThat(scmConfigurationView.fields().size(), is(3));
+        assertThat(scmConfigurationView.fields().size(), is(4));
     }
 
     @Test

